lua/autorun/nl_library.lua

```diff
@@ -0,0 +1,609 @@
+			
+			ply.nllthirdpersonview = not ply.nllthirdpersonview
+		end
+	end
+end)
+local dist = 200
+local view = {}
+hook.Add("CalcView", "TG_NLL", function(ply, origin, angles, fov)
+	local rag = ply:GetNWEntity("nllviewrag")
+	if IsValid(rag) then
+		local bid = rag:LookupBone("ValveBiped.Bip01_Head1")
+		if bid then
+			local dothirdperson = false
+			dothirdperson = ply.nllthirdpersonview
+			
+			if dothirdperson then
+				local ragpos = rag:GetBonePosition(bid)
+				
+				local pos = ragpos - (ply:GetAimVector()*dist)
+				local ang = (ragpos - pos):Angle()
+				
+				--Do a traceline so he can't see through walls
+				local trdata = {}
+				trdata.start = ragpos
+				trdata.endpos = pos
+				trdata.filter = rag
+				local trres = util.TraceLine(trdata)
+				if trres.Hit then
+					pos = trres.HitPos + (trres.HitWorld and trres.HitNormal * 3 or vector_origin)
+				end
+				
+				view.origin = pos
+				view.angles = ang
+			else
+				local pos,ang = rag:GetBonePosition(bid)
+				pos = pos + ang:Forward() * 7
+				ang:RotateAroundAxis(ang:Up(), -90)
+				ang:RotateAroundAxis(ang:Forward(), -90)
+				pos = pos + ang:Forward() * 1
+				
+				view.origin = pos
+				view.angles = ang
+			end
+			
+			return view
+		end
+	end
+end)
+local w,h = ScrW(), ScrH()
+local w2,h2 = w/2,h/2
+local function IsOnScreen(pos)
+	return pos.x > 0 and pos.x < w and pos.y > 0 and pos.y < h
+end
+local function GrabPlyInfo(ply)
+
+	local text, color
+	if ply:GetNWBool("disguised", false) then
+		if LocalPlayer():IsTraitor() or LocalPlayer():IsSpec() then
+			text = ply:Nick() .. LANG.GetUnsafeLanguageTable().target_disg
+		else
+			-- Do not show anything
+			return
+		end
+		color = COLOR_RED
+	else
+		text = ply:Nick()
+	end
+	
+	return text, (color or COLOR_WHITE), "TargetID"
+end
+
+function SimpleTextShadowed(txt, font, wpos, hpos, color, size)
+	draw.SimpleText(txt, font, wpos - 1, hpos - 1, Color(0,0,0,255), size)
+	draw.SimpleText(txt, font, wpos, hpos, color, size)
+end
+
+hook.Add("HUDPaint", "TG_NLL", function()
+	--Draws info about crouch able to switch between third and firstperson
+	if IsValid(LocalPlayer():GetNWEntity("nllviewrag")) then
+		local txt = string.format("Press %s to switch between third and firstperson view.", input.LookupBinding("+duck"))
+		SimpleTextShadowed(txt, "TargetID", w2, h/10, Color(200,200,200,255), 1)
+	end
+	
+	local targ = LocalPlayer():GetEyeTrace().Entity
+	if IsValid(targ) and IsValid(targ:GetNWEntity("nllplyowner")) and LocalPlayer():GetPos():Distance(targ:GetPos()) < 400 then
+		local pos = targ:GetPos():ToScreen()
+		if IsOnScreen(pos) then
+			local ply = targ:GetNWEntity("nllplyowner")
+			local nick,nickclr,font = GrabPlyInfo(ply)
+			if not nick then return end -- Someone doesn't want us to draw his info.
+			
+			SimpleTextShadowed(nick, font, pos.x, pos.y - 50, nickclr, 1)
+			if not condcolors[ply.condition] then conditionColor = Color(150,150,255) else conditionColor = condcolors[ply.condition][1] end
```

Comment:
Ditto.

Reply:
This is the clientside code, which does not need to loop through each conditioned player. Since I am not going to be using a table, I won't be defining a table.

Reply:
Ditto as other comment.